ginga/AstroImage.py

```diff
@@ -106,41 +97,13 @@ def setup_data(self, data, naxispath=None):
         self.set_naxispath(naxispath)
 
     def load_hdu(self, hdu, fobj=None, naxispath=None,
-                 inherit_primary_header=None):
-
-        # this seems to be necessary now for some fits files...
-        try:
-            hdu.verify('fix')
-
-        except Exception as e:
-            # Let's hope for the best!
-            self.logger.warning("Problem verifying fits HDU: {}".format(e))
-
-        self.clear_metadata()
-
-        # collect HDU header
-        ahdr = self.get_header()
-        self._copy_hdu_header(hdu, ahdr)
-
-        # Set PRIMARY header
-        if inherit_primary_header is None:
-            inherit_primary_header = self.inherit_primary_header
-        else:  # This ensures get_header() is consistent
-            self.inherit_primary_header = inherit_primary_header
+                 save_primary_header=False):
```

Comment:
Same comment as https://github.com/ejeschke/ginga/pull/978/files#r776394464 , except the warning here will be a bit different (renaming instead of removing).

`astropy` supports both cases: See https://github.com/astropy/astropy/blob/main/astropy/utils/decorators.py